prov/verbs/src/fi_verbs.c

```diff
@@ -3177,12 +3221,26 @@ fi_ibv_domain(struct fid_fabric *fabric, struct fi_info *info,
 	   struct fid_domain **domain, void *context)
 {
 	struct fi_ibv_domain *_domain;
+	struct fi_info *fi;
 	int ret;
 
+	fi = fi_ibv_find_info(NULL, info->domain_attr->name);
+	if (!fi)
+		return -FI_EINVAL;
+
+	ret = fi_ibv_check_domain_attr(info->domain_attr, fi);
+	if (ret)
+		return ret;
+
 	_domain = calloc(1, sizeof *_domain);
 	if (!_domain)
 		return -FI_ENOMEM;
 
+	if (!(_domain->info = fi_dupinfo(fi))) {
+		ret = -FI_ENOMEM;
+		goto err;
+	}
```

Comment:
The fi_info (fi) stored with domain here would be used to check the fi_info passed by the user in fi_endpoint call. 

fi is an element in verbs_info list. I'm making a copy of it so domain accesses only what it needs. fi_dupinfo copies just the element and not the entire list. This might be a little too much caution. Please let me know if that's not needed.